File: ios/KeyChainWrapper.m
//
//  KeyChainWrapper.m
//  DemoApp
//
//  Created by Amanpreet Singh on 12/03/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(KeyChainWrapper, NSObject)
RCT_EXTERN_METHOD(savePassword:(NSString*)password callback:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getPassword:(RCTResponseSenderBlock)callback)

@end

File: ios/KeyChainWrapper.swift
//
//  KeyChainWrapper.swift
//  DemoApp
//
//  Created by Amanpreet Singh on 12/03/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import SwiftKeychainWrapper

@objc(KeyChainWrapper)
class KeyChainWrapper: NSObject {
  
  @objc
     func savePassword(_ password:String,  callback:RCTResponseSenderBlock){
        if(KeychainWrapper.standard.set(password, forKey: "password")){
          callback([true])
        }
        else{
          callback([false])
        }
    }
  
  @objc
   func getPassword(_ callback:RCTResponseSenderBlock){
    if let passwod = KeychainWrapper.standard.string(forKey: "password"){
        callback([passwod])
      }
      else{
        callback([NSNull()])
      }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
